include/chunk/region.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "chunk/chunk.h"
+#include <memory>
+
+namespace chunk
+{
+
+/**
+ * Class containing hash function for std::unordered_map
+ */
+struct ChunkHasher
+{
+    std::size_t operator()( const pos::ChunkPos& chunk_pos ) const
+    {
+        std::hash<int64_t> hasher{};
+        return hasher( int64_t( chunk_pos ) );
+    }
+};
+
+/**
+ * Class that contains all block ides and manage the chunks
+ */
+
+class Region
+{
+  public:
+    using ChunkMap = std::unordered_map<pos::ChunkPos, Chunk, ChunkHasher>;
+    using BlockIDPtr = std::unique_ptr<BlockID[]>;
+
+  public:
+    Region( const Region& ) = delete;
+    Region( Region&& ) = delete;
+
+    Region& operator=( const Region& ) = delete;
+    Region& operator=( Region&& ) = delete;
+    ~Region() = default;
+
+    static Region& getRef()
+    {
+        static Region singleton_region{ { 0, 0 } };
+        return singleton_region;
+    }
+
+    Chunk& getChunk( const pos::ChunkPos& pos );
+    pos::ChunkPos& getBeginPos() { return m_begin_pos; }
+    pos::ChunkPos getBeginPos() const { return m_begin_pos; }
```

Comment:
I suppose there is no difference, anyway it will be copied

Reply:
If you use return value as an argument of a function, that takes `const pos::ChunkPos&`, it won't be copied.

Reply:
Well if you assign it to a reference yes, but if for object, that used in most case, it will be copied. Anyway, the ChunkPos light-weight class, so i don't understand why it should be returned by reference.

Reply:
Okay the point about function, that takes `const pos::ChunkPos&` it won't be copied is true, I will resolve it.